core/common/src/main/java/alluxio/wire/MasterInfo.java

```diff
@@ -0,0 +1,153 @@
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.wire;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Master information.
+ */
+@NotThreadSafe
+public final class MasterInfo {
+  /** Master's address. */
+  private Address mAddress;
+  /** The id of the master. */
+  private long mId;
+  /** Master's last updated time in ms. */
+  private long mLastUpdatedTimeMs;
+
+  /**
+   * Creates a new instance of {@link MasterInfo}.
+   */
+  public MasterInfo() {}
+
+  /**
+   * Creates a new instance of {@link MasterInfo}.
+   *
+   * @param id the master id to use
+   * @param address the master address to use
+   * @param lastUpdatedTimeMs the master lastUpdatedTimeMs to use
+   */
+  public MasterInfo(long id, Address address, long lastUpdatedTimeMs) {
+    mAddress = Preconditions.checkNotNull(address, "address");
+    mId = id;
+    mLastUpdatedTimeMs = lastUpdatedTimeMs;
+  }
+
+  /**
+   * Creates a new instance of {@link MasterInfo}.
+   *
+   * @param id the master id to use
+   * @param address the master address to use
+   */
+  public MasterInfo(long id, Address address) {
+    mAddress = Preconditions.checkNotNull(address, "address");
+    mId = id;
+    mLastUpdatedTimeMs = System.currentTimeMillis();
+  }
+
+  /**
+   * Creates a new instance of {@link MasterInfo}.
+   *
+   * @param masterInfo the masterInfo need to be cloned
+   * @return the the master information
+   */
+  public MasterInfo clone(MasterInfo masterInfo) {
+    mAddress = masterInfo.getAddress();
+    mId = masterInfo.getId();
+    mLastUpdatedTimeMs = masterInfo.getLastUpdatedTimeMs();
+    return this;
+  }
+
+  /**
+   * @return the master's address
+   */
+  public Address getAddress() {
+    return mAddress;
+  }
+
+  /**
+   * @return the id of the master
+   */
+  public long getId() {
+    return mId;
+  }
+
+  /**
+   * @return the last updated time of the master in ms
+   */
+  public long getLastUpdatedTimeMs() {
+    return mLastUpdatedTimeMs;
+  }
+
+  /**
+   * @param address the master address information
+   * @return the master information
+   */
+  public MasterInfo setAddress(Address address) {
+    mAddress = address;
+    return this;
+  }
+
+  /**
+   * @param id the master id
+   * @return the master information
+   */
+  public MasterInfo setId(long id) {
+    mId = id;
+    return this;
+  }
+
+  /**
+   * @param lastUpdatedTimeMs the last update time in ms
+   * @return the master information
+   */
+  public MasterInfo setLastUpdatedTimeMs(long lastUpdatedTimeMs) {
+    mLastUpdatedTimeMs = lastUpdatedTimeMs;
+    return this;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this).add("id", mId).add("address", mAddress)
+        .add("lastUpdatedTimeMs", mLastUpdatedTimeMs).toString();
```

Comment:
convert the MS to better format with `CommonUtils.convertMsToClockTime(mLastUpdatedTimeMs)`